dbms/src/Core/Settings.h

```diff
@@ -219,9 +219,20 @@ struct Settings : public SettingsCollection<Settings>
     M(SettingInt64, distributed_ddl_task_timeout, 180, "Timeout for DDL query responses from all hosts in cluster. If a ddl request has not been performed on all hosts, a response will contain a timeout error and a request will be executed in an async mode. Negative value means infinite.") \
     M(SettingMilliseconds, stream_flush_interval_ms, 7500, "Timeout for flushing data from streaming storages.") \
     M(SettingMilliseconds, stream_poll_timeout_ms, 500, "Timeout for polling data from/to streaming storages.") \
+    \
     M(SettingString, format_schema, "", "Schema identifier (used by schema-based formats)") \
-    M(SettingString, format_schema_rows, "", "Row format string for Template format") \
-    M(SettingString, format_schema_rows_between_delimiter, "\n", "Delimiter between rows for Template format") \
+    M(SettingString, format_template_resultset, "", "Path to file which contains format string for result set (for Template format)") \
+    M(SettingString, format_template_row, "", "Path to file which contains format string for rows (for Template format)") \
+    M(SettingString, format_template_rows_between_delimiter, "\n", "Delimiter between rows (for Template format)") \
+    \
+    M(SettingString, format_custom_escaping_rule, "", "Field escaping rule (for CustomSeparated format)") \
+    M(SettingString, format_custom_field_delimiter, "\t", "Delimiter between fields (for CustomSeparated format)") \
+    M(SettingString, format_custom_row_before_delimiter, "", "Delimiter before field of the first column (for CustomSeparated format)") \
+    M(SettingString, format_custom_row_after_delimiter, "", "Delimiter after field of the last column (for CustomSeparated format)") \
+    M(SettingString, format_custom_row_between_delimiter, "\n", "Delimiter between rows (for CustomSeparated format)") \
```

Comment:
Probably we want to move \n from `format_custom_row_between_delimiter` to `format_custom_row_after_delimiter` by default to correspond to TSV.